### PR TITLE
Send author requests via `QueuedEmail`

### PR DESCRIPTION
**Pull request by nimmolo**
Just what it says.

### CODE REVIEW

File: app/models/queued_email.rb [hunk 1]
```diff
@@ -129,18 +129,6 @@ class QueuedEmail < AbstractModel
...
-  # loaded, so it won't know to try to load the one in QueuedEmail.  This way,
-  # soon as QueuedEmail is defined, we know that all subclasses are also
-  # properly defined, and we no longer have to rely on autoloading.
-  #
-  Dir["#{::Rails.root}/app/models/queued_email/*.rb"].each do |file|
-    match = /(\w+)\.rb$/.match(file)
-    require("queued_email/#{match[1]}") if match
-  end
-
   # ----------------------------
```
Comment (nimmolo) [inline app/models/queued_email.rb:132]: This is not necessary because of zeitwerk.

**Comment by nimmolo** [comment]
Note that this is only a PR off the `AuthorsController` PR, merging it will only combine this with that PR.

File: app/models/queued_email.rb (hunk 1 above)
Comment (pellaea) [reply app/models/queued_email.rb:132]: Nice!

File: app/models/queued_email/author_request.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  def self.create_email(sender, recipient, object, subject, message)
+    result = create(sender, recipient)
+    raise("Missing object!") unless object
+
+    result.add_integer(:obj_id, object.id)
+    result.add_string(:obj_type, object.type_tag.to_s)
+    result.add_string(:subject, subject)
+    result.set_note(message)
+    result.finish
+    result
```
Comment (pellaea) [inline app/models/queued_email/author_request.rb:30]: Oooh!  I just thought of something.  What if QueuedEmail#add_xxx, set_note, finish all returned self?  Then these could all be chained:

create().add_xxx(xxx).add_xxx(xxx).set_note(xxx).finish

That would be so slick! :)

**Comment by pellaea** [comment]
Wow, that was easy.  Maybe I'll do that with some of the other non-QueuedEmail mail, too.  Once all of our email uses this mechanism, then we can contemplate alternative delivery methods, something that's been near the top of my personal priority list for ages.

Looks good to me.  Thanks for doing this and demonstrating how easy it is to migrate direct emails to QueuedEmails.

File: app/models/queued_email/author_request.rb (hunk 2 above)
Comment (nimmolo) [reply app/models/queued_email/author_request.rb:30]: The `return unless object` currently interrupts that chain, should it come before?

Comment (nimmolo) [reply app/models/queued_email/author_request.rb:30]: Trying it.  Like so?
```
    raise("Missing object!") unless object

    create(sender, recipient).
      add_integer(:obj_id, object.id).
      add_string(:obj_type, object.type_tag.to_s).
      add_string(:subject, subject).
      set_note(message).finish
```

**Comment by pellaea** [comment]
Yes, that's _exactly_ what I mean.  It's silly syntactical sugar, but what
can I say, I'm just in that mood right now. :)  Please feel free to ignore!

**Comment by nimmolo** [comment]
`add_integer` actually throws an error chained immediately after `create` - I think it has to wait for the minimal record to be instantiated. I'll split it into create/chain

**Pull request merged**